tests/models/flava/test_modeling_flava.py

```diff
@@ -1338,3 +1338,48 @@ def test_inference(self):
         self.assertAlmostEqual(outputs.loss_info.mmm_text.item(), 1.75533199, places=4)
         self.assertAlmostEqual(outputs.loss_info.mmm_image.item(), 7.0290069, places=4)
         self.assertAlmostEqual(outputs.loss.item(), 11.0626, places=4)
+
+    @slow
+    def test_inference_with_itm_labels(self):
+        model_name = "facebook/flava-full"
+        model = FlavaForPreTraining.from_pretrained(model_name).to(torch_device)
+        processor = FlavaProcessor.from_pretrained(model_name)
+        torch.manual_seed(1)
+        random.seed(1)
+
+        image = prepare_img()
+        inputs = processor(
+            text=["a photo of a cat", "a photo of a dog"],
+            images=[image, image],
+            padding="max_length",
+            max_length=77,
+            return_tensors="pt",
+            return_codebook_pixels=True,
+            return_image_mask=True,
+        )
+        inputs["input_ids_masked"] = inputs["input_ids"].clone()
+        inputs["input_ids_masked"][0, 4:6] = 103
+        inputs["mlm_labels"] = inputs["input_ids"].clone()
+        inputs["mlm_labels"][:, :] = -100
+        inputs["mlm_labels"][0, 4:6] = inputs["input_ids"][0, 4:6]
+        inputs["itm_labels"] = torch.tensor([1, 0])  # first pair matches, second one doesn't
```

Comment:
Let's rather declare each one of these explicitly! 
this way we can "see" what's going on!

Reply:
That's the only thing left!

Reply:
@ArthurZucker I have added comments to better explain the process. I did not explicitly initialize the tensors because I am following the same procedure as the "test_inference" test on line 1271. Additionally, I have included comments in that test (from which I drew inspiration).

If you want to still make it explicit, we would need to update that test too